### PR TITLE
Formatting page

### PR DESCRIPTION
**Pull request by bgavas**

### CODE REVIEW

**Comment by bgavas** [comment]
Formatting conversion is outstanding for this ticket
Screen Shot 2022-11-29 at 21 19 31

Date formats should be created from enums

File: packages/calendar/core/src/constants.ts [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+// @ts-ignore
+import flagDeIcon from "@agreeto/ui/src/assets/flag-de.svg";
+// @ts-ignore
+import flagEsIcon from "@agreeto/ui/src/assets/flag-es.svg";
+// @ts-ignore
+import flagFrIcon from "@agreeto/ui/src/assets/flag-fr.svg";
+// @ts-ignore
+import flagItIcon from "@agreeto/ui/src/assets/flag-it.svg";
+// @ts-ignore
+import flagUSIcon from "@agreeto/ui/src/assets/flag-us.svg";
```
Comment (p6l-richard) [inline packages/calendar/core/src/constants.ts:12]: These disables/ignores can be avoided when you export a react component from the `@agreeto/ui` package
CleanShot 2022-11-29 at 21 39 46@2x

Basically like we're [already doing](https://github.com/p6l-richard/create-t3-turbo/blob/19b4aff33e0a58b1dfc6fe9ce80bdc59bd42e3da/packages/ui/src/logo/google.tsx#L3) for the other icons.

File: packages/calendar/core/src/interfaces/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import type { Language } from "@prisma/client";
```
Comment (p6l-richard) [inline packages/calendar/core/src/interfaces/index.ts:1]: We can co-locate this to the `calendar-core/src/constants.ts` file so that we we update the constant we have it's interface in the same place.

File: packages/calendar/react/src/components/action-pane/availability.tsx [hunk 3]
```diff
@@ -172,78 +205,90 @@ const Availability: React.FC<{
...
+                {formatting?.introSentence ??
+                getCopyTitle(preference?.formatLanguage) ? (
+                  <div className="text-sm" style={{ paddingTop: "2px" }}>
+                    <span className="color-gray-700">
+                      {formatting?.introSentence ??
+                        getCopyTitle(preference?.formatLanguage)}
+                    </span>
+                  </div>
+                ) : (
+                  <div />
```
Comment (p6l-richard) [inline packages/calendar/react/src/components/action-pane/availability.tsx:230]: 😅

File: packages/calendar/react/src/components/action-pane/availability.tsx [hunk 4]
```diff
@@ -172,78 +205,90 @@ const Availability: React.FC<{
...
+            </div>
+          ) : (
+            <div>
+              <div className="flex justify-between pb-3">
+                {formatting?.introSentence ??
+                getCopyTitle(preference?.formatLanguage) ? (
+                  <div className="text-sm" style={{ paddingTop: "2px" }}>
+                    <span className="color-gray-700">
+                      {formatting?.introSentence ??
+                        getCopyTitle(preference?.formatLanguage)}
```
Comment (p6l-richard) [inline packages/calendar/react/src/components/action-pane/availability.tsx:226]: Looks like it's the same as in lines [221-222](https://github.com/agreeto/agreeto/pull/15/files#diff-ae40774dc406ba05e0a9b78d9622fa6921af1219c18a951efc612a1bd795d584R221-R222) and [238-239](https://github.com/agreeto/agreeto/pull/15/files#diff-ae40774dc406ba05e0a9b78d9622fa6921af1219c18a951efc612a1bd795d584R238-R239)?

could extract to constants to improve readbility

File: packages/db/prisma/schema.prisma [hunk 5]
```diff
@@ -170,6 +188,21 @@ model Preference {
     user User @relation(fields: [userId], references: [id])
 }
 
+model Formatting {
```
Comment (p6l-richard) [inline packages/db/prisma/schema.prisma:191]: I like that this table is `formatting` only 👍

**Review by p6l-richard — COMMENTED** [review]
gave it a first view and left some comments here + started some discussion on Slack if you're up for it. 

Will check out branch tomorrow to give it a spin. :)

File: packages/calendar/core/src/constants.ts (hunk 1 above)
Comment (juliusmarminge) [reply packages/calendar/core/src/constants.ts:12]: I went for a lib for these but we could create components like the GoogleIcon for these too

https://github.com/agreeto/agreeto/blob/7ff64b9dfed4fb34dde184b9870ea1311624ddd7/packages/calendar/react/src/components/action-pane/availability.tsx#L66-L69

or 

https://github.com/agreeto/agreeto/blob/7ff64b9dfed4fb34dde184b9870ea1311624ddd7/packages/ui/src/logo/google.tsx#L3

**Comment by bgavas** [comment]
@p6l-richard @juliusmarminge I got millions of conflicts. Can anybody handle it since you are more familiar with the current codebase? Idk which one to choose in most cases

**Comment by juliusmarminge** [comment]
Resolved conflicts and did the flag thing

File: apps/extension/src/pages/formatting.tsx [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+      setIntroSentenceType(formatting.introSentenceType);
+      setIntroSentence(formatting.introSentence);
+    } else {
+      // Reset fields
+      setSelectedDateFormat(DateFormat.MMMM_d_EEEE);
+      setIntroSentenceType(IntroSentenceType.DEFAULT);
+      resetIntroSentence(IntroSentenceType.DEFAULT);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedLanguage, formattings]);
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:67]: This is a bit sketchy imo, no need to make all of this state and synchronizing in an effect. We can inherit these values from the formatting as regular variables

It's sort of similar to the antipattern:
```ts
const [firstName, setFirstName] = useState("")
const [lastName, setLastName] = useState("")
const [fullName, setFullName] = useState(firstName + lastName)

useEffect(() => {
  setFullName(firstName + lastName)
}, [firstName, lastName])
```

where you could just 
```ts
const [firstName, setFirstName] = useState("")
const [lastName, setLastName] = useState("")
const fullName = firstName + lastName
```

File: apps/extension/src/pages/formatting.tsx [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+  const { data: formattings } = trpcApi.formatting.byCurrentUser.useQuery();
+  const { mutate: updateFormatting, isLoading: isUpdating } =
+    trpcApi.formatting.update.useMutation({
+      onSuccess() {
+        utils.formatting.byCurrentUser.invalidate();
+      },
+    });
+
+  useEffect(() => {
+    const formatting = formattings?.find(
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:48]: trpc procedures are cheap, can make a separate procedure like `byLanguage` and just send the language in, and select a single formatting from the db that match the language there, instead of sending all of them back and filtering here

File: apps/extension/src/pages/formatting.tsx [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+import { trpcApi } from "~features/trpc/api/hooks";
+
+// @ts-expect-error - svg's are not typed
+import previewImage from "./../../assets/gmail-preview.svg";
+
+export const Formatting = () => {
+  const utils = trpcApi.useContext();
+  const introSentenceRef = useRef<HTMLTextAreaElement>(null);
+  const [languageDropdownOpen, setLanguageDropdownOpen] = useState(false);
+  const [dateFormatDropdownOpen, setDateFormatDropdownOpen] = useState(false);
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:23]: what are these used for? radix handles the open/close state itself?

File: apps/extension/src/pages/formatting.tsx [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+  );
+
+  const introSentenceElem = (
+    <div>
+      <div className="flex border border-gray-100 rounded w-full h-9 overflow-hidden">
+        {[
+          IntroSentenceType.DEFAULT,
+          IntroSentenceType.CUSTOM,
+          IntroSentenceType.NONE,
+        ].map((type) => (
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:131]:
```suggestion
        {Object.values(IntroSentenceType)
          .map((type) => (
```

File: apps/extension/src/pages/formatting.tsx [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+        {[
+          DateFormat.MMMM_d_EEEE,
+          DateFormat.MM_dd_yyyy,
+          DateFormat.yyyy_MM_dd,
+          DateFormat.MMMM_dd_yyyy,
+          DateFormat.EEEE_MM_dd_yyyy,
+          DateFormat.MMM_dd_EEEE,
+          DateFormat.EEEE_M_d,
+          DateFormat.EEE_MM_dd,
+        ].map((item) => (
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:204]:
```suggestion
        {Object.values(DateFormat)
          .map((item) => (
```

File: apps/extension/src/pages/formatting.tsx [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+              ],
+              {
+                formatLanguage: selectedLanguage.key,
+                dateFormat: selectedDateFormat,
+                copyTitle: introSentence,
+                highlight: {
+                  date: languageDropdownOpen || dateFormatDropdownOpen,
+                  introSentence: languageDropdownOpen || introSentenceFocused,
+                  time: languageDropdownOpen,
+                },
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:279]: what is actually highlighting here? i didn't see any difference in the ui

**Review by juliusmarminge — COMMENTED** [review]
We could look into react-hook-form, react-zorm or some other form library to handle a lot of the stuff here.

Right now you're handling all the refs, if the values have changed, whether a field is focused or not etc etc yourself which can become cumbersome if it grows more complex.

react-hook-form is pretty nice along with trpc: https://kitchen-sink.trpc.io/react-hook-form?file=feature%2Freact-hook-form%2Findex.tsx#content

File: apps/extension/src/pages/formatting.tsx (hunk 6 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:67]: Yeah would be cool if we can make this part more clear. But what do you suggest? I didn't get it.

File: apps/extension/src/pages/formatting.tsx (hunk 7 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:48]: I would prefer no delay there for UX. What do you think @p6l-richard? Correct me if I am wrong but as far as I understand, API calls take longer when made from extension. Like when you change the language by dropdown on top of the availability text-area, it takes too long. Idk why.

This is the dropdown place in old version
Screen Shot 2022-12-01 at 10 16 35

File: apps/extension/src/pages/formatting.tsx (hunk 8 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:23]: They are for highlighting parts in the preview. I copied and pasted from the old repo and apparently some CSS are broken. Normally those states update specific text color like in the screenshot taken from old app:

Screen Shot 2022-12-01 at 10 19 30
Screen Shot 2022-12-01 at 10 19 47

File: apps/extension/src/pages/formatting.tsx (hunk 9 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:131]: Good catch. I thought `Object.values` will break the order but apparently it doesn't

File: apps/extension/src/pages/formatting.tsx (hunk 11 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:268]: Mentioned [here](https://github.com/agreeto/agreeto/pull/15#discussion_r1036764201)

File: apps/extension/src/pages/formatting.tsx (hunk 7 above)
Comment (bgavas) [reply apps/extension/src/pages/formatting.tsx:48]: So, I had to stop my work in this repo and started to work on the old one to complete the formatting page completely. What is missing is the `Advanced Formatting` feature. After I complete that part, I will be update this PR with CRs

File: apps/extension/src/pages/formatting.tsx (hunk 6 above)
Comment (p6l-richard) [reply apps/extension/src/pages/formatting.tsx:67]: That's a great discussion here that adds a lot of value to our engineering. I appreciate both comments. 

The tl;dr of my comment:
- too many useState/useEffect's are a sign of handling react state inefficiently leading to bugs and increased maintainability
- given how fundamental our dependency on react is & best practices being readily available, we should spend the time to fix this now and not some time in the future

And here's the actual content 👇

I agree that the usage of these useEffect to sync states is a code smell to me. 

I'm also not sure specifically what you're referring to @juliusmarminge, but the best reference on deciding to useEffect or not for me are Dan abramov's docs on it, which are the official react beta docs. 

Eg, it looks like we could handle the state in an event handler instead.

image


Another point that's stated is to not useState at all, where possible thereby passing the need to sync state between each other. 
image
image

Lastly, I think we should also not reset state based on prop changes if we can avoid that, as per the docs:
image
image

image

For form data specifically, I think state is overused. It's possible to use the web browser's API for this and remix demonstrated this and some react form libraries take the same approach: use browser api's over custom and unnecessary react abstractions. The big advantage is that react is based on JS with this in mind, so it plays together quite nicely. 

Eg consider this form that doesn't event need a hook nor a form library:
image
image
[Source](https://twitter.com/DavidKPiano/status/1569361453928300545?s=20&t=ttjxrYx4pkH7F5YwaISFPQ) 


That's what I like about react-zorm:
It uses browser APIs where possible, but adds Typesafety where useful. An example for this is the event.target type, which is an issue in TypeScript with FormData. You can use zod to make it type safe, which is basically what react-zorm does for you as a lib conceptionally. 

All of these best practices are to use react's data flow for state more efficiently. 

While I'm aware that understanding react's state flow throughout the component cycle is hard to understand, I think making an effort to learn this pays off in the future from my personal experience. Adding state and useEffects leads to components which are very hard to debug because it's not always straightforward to understand the data flow of state in react components. Therefore, if we can keep it as vanilla and follow react's recommendations as much as possible, we will improve maintainability. 

If we do encounter questions about this in the future, I found that [Dan was also responsive](https://twitter.com/dan_abramov/status/1541821478009798658?s=20&t=eGg9MO7Wip7l3251o16ujg) into looking into specific use cases but I think here it just pays off to follow React's recommended practices for when not to useEffect. That should already clean up quite a bit.

Comment (p6l-richard) [reply apps/extension/src/pages/formatting.tsx:67]: Having said that, I do believe that there's. Big value in tying our form library into our router. 

Remix has demonstrated how nicely this enables optimistic UI without the downsides. 

Given that trpc enables us to invalidate not only based on routes but other calls on top of that, I'd be happy with a form<>tRPC binding but would honestly prefer a router<>form binding that is put on steroids by using tRPC. 

That's my take.

File: apps/extension/src/pages/formatting.tsx (hunk 7 above)
Comment (p6l-richard) [reply apps/extension/src/pages/formatting.tsx:48]: I agree with @bgavas. We have to find a way to not delay the UX. 

I still think that this should be handled server-side though as this is our custom "availability appearance" logic and doesn't have anything to do with the react-calendar per se. Therefore it does feel misplaced somewhat (I am responsible for the location of this in the calendar-react package by the way). 

Since we don't have an @agreeto/availability-core and @agreeto/availability-react package yet, I think putting this on our server is acceptable. 

However, we need to find a way not to delay the UX. Therefore, I think we need to prefetch this during the hydration of our app. Whenever the user opens the react app for the first time, we could fetch this data along with other required data on the calendar route (eg events, verify session etc). 

The downside I see with an approach like this is that if we're not careful about structuring our app properly in our routes, we may run into waterfalls of a lot of requests happening. 
Since we're using tan stack router and have trpc batching implemented, I think we're well equipped to avoid the waterfalls.

We just need to pay attention to not stuff too many requests into our hydration over time. 

The good thing: this data needs to be fetched only once per component lifetime and won't change when changing weeks for example. So I do consider it cheaper than the events request and would find the server-side location placement acceptable if we find a way not to delay the UX for loading the options. 

The same applies to the co worker search bar by the way. 

I think we could prefetch user.getFriends ahead of time by simply reading common friends from the events of the current and/or past week. 
Out of scope of this PR though.

File: apps/extension/src/pages/formatting.tsx (hunk 6 above)
Comment (juliusmarminge) [reply apps/extension/src/pages/formatting.tsx:67]: The name example was somewhere in those screenshots @p6l-richard posted.

Essentially, any data that is inherited from another state shouldn't be state in itself. In our case this would be the 
```ts
 const [introSentence, setIntroSentence] = useState<string>(
     DEFAULT_LANGUAGE_FORMAT.defaultIntroSentence,
   );
   const [introSentenceType, setIntroSentenceType] = useState<IntroSentenceType>(
     IntroSentenceType.DEFAULT,

useEffect(() => {
  // update when formatting changes
}, [formatting]);
```
to 
```ts
const introSentence = formatting?.introSentence ?? DEFAULT_LANGUAGE_FORMAT.defaultIntroSentence;
const introSentenceType = formatting?.introSentenceType ?? IntroSentenceType.DEFAULT
```

No need to synchronize them in an effect.

For stuff that still would need synchronization, the useQuery's onSuccess, onError handlers are better here in the sense that we use the primitives that our state-provider gives us.

I don't know if that made it clearer?

File: apps/extension/src/pages/formatting.tsx [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+      if (lang) {
+        setIntroSentence(lang.defaultIntroSentence);
+      }
+    }
+  };
+
+  const languageDropdown = (
+    <DropdownMenu.Root onOpenChange={setLanguageDropdownOpen}>
+      <DropdownMenu.Trigger>
+        <div className="h-10 bg-white color-gray-700 rounded border border-gray-200 w-64 flex py-1 px-3 justify-between items-center space-x-3 text-sm">
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:85]:
```suggestion
        <div className="h-10 bg-white text-gray-700 rounded border border-gray-200 w-64 flex py-1 px-3 justify-between items-center space-x-3 text-sm">
```

File: apps/extension/src/pages/formatting.tsx [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+            <selectedLanguage.icon className="h-5 w-5" />
+            <div>{selectedLanguage.title}</div>
+          </div>
+
+          <TiArrowSortedDown className="h-6 w-6 text-gray-600" />
+        </div>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content
+        className="z-10 bg-white color-gray-700 rounded mt-2"
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:96]:
```suggestion
        className="z-10 bg-white text-gray-700 rounded mt-2"
```

File: apps/extension/src/pages/formatting.tsx [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+        {LANGUAGE_FORMATS.map((item) => (
+          <DropdownMenu.Item
+            key={item.key}
+            onSelect={() => {
+              if (item.key === selectedLanguage.key) return;
+              setSelectedLanguage(item);
+              setCanSave(false);
+            }}
+          >
+            <div className="h-10 bg-white color-gray-700 w-64 flex py-1 px-3 justify-between space-x-3 cursor-pointer text-sm">
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:109]:
```suggestion
            <div className="h-10 bg-white text-gray-700 w-64 flex py-1 px-3 justify-between space-x-3 cursor-pointer text-sm">
```

File: apps/extension/src/pages/formatting.tsx [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+    <div>
+      <div className="flex border border-gray-100 rounded w-full h-9 overflow-hidden">
+        {Object.values(IntroSentenceType)
+          .map((type) => (
+          <div
+            key={type}
+            className={`flex-1 cursor-pointer text-sm flex items-center justify-center ${
+              introSentenceType === type
+                ? "bg-primary text-white"
+                : "bg-white color-[#3A3F46] hover:bg-gray-100"
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:134]:
```suggestion
                : "bg-white text-[#3A3F46] hover:bg-gray-100"
```

File: apps/extension/src/pages/formatting.tsx [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+          }}
+        />
+      </div>
+    </div>
+  );
+
+  const dateFormatDropdown = (
+    <DropdownMenu.Root onOpenChange={setDateFormatDropdownOpen}>
+      <DropdownMenu.Trigger>
+        <div className="h-10 bg-white color-gray-700 rounded border border-gray-200 w-64 flex py-1 px-3 justify-between space-x-3 items-center text-sm">
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:181]:
```suggestion
        <div className="h-10 bg-white text-gray-700 rounded border border-gray-200 w-64 flex py-1 px-3 justify-between space-x-3 items-center text-sm">
```

File: apps/extension/src/pages/formatting.tsx [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+          <div>{format(new Date(), selectedDateFormat)}</div>
+          <TiArrowSortedDown className="h-6 w-6 text-gray-600" />
+        </div>
+      </DropdownMenu.Trigger>
+
+      <DropdownMenu.Content
+        className="z-10 bg-white color-gray-700 rounded mt-2"
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:188]:
```suggestion
        className="z-10 bg-white text-gray-700 rounded mt-2"
```

File: apps/extension/src/pages/formatting.tsx [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+          .map((item) => (
+          <DropdownMenu.Item
+            key={item}
+            onSelect={() => {
+              if (item === selectedDateFormat) return;
+              setSelectedDateFormat(item);
+              setCanSave(true);
+            }}
+          >
+            <div className="h-10 bg-white color-gray-700 w-64 flex py-1 px-3 justify-between space-x-3 cursor-pointer items-center text-sm">
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:202]:
```suggestion
            <div className="h-10 bg-white text-gray-700 w-64 flex py-1 px-3 justify-between space-x-3 cursor-pointer items-center text-sm">
```

File: packages/calendar/react/src/components/action-pane/availability.tsx [hunk 19]
```diff
@@ -173,78 +206,90 @@ const Availability: React.FC<{
...
-          </div>
-          {Object.keys(getGroupedSlots(selectedSlots)).map((key, idx) => {
-            const events: EventInput[] = getGroupedSlots(selectedSlots)[key];
-            const firstEvent = events[0];
-            if (!firstEvent) return null;
+          {!selectedSlots.length ? (
+            <div className="flex justify-between">
+              <div
+                className="color-gray-200 text-sm"
+                style={{ paddingTop: "2px" }}
```
Comment (juliusmarminge) [inline packages/calendar/react/src/components/action-pane/availability.tsx:213]:
```suggestion
                className="text-gray-200 text-sm pt-[2px]"
```

File: packages/calendar/react/src/components/action-pane/availability.tsx [hunk 20]
```diff
@@ -173,78 +206,90 @@ const Availability: React.FC<{
...
+              </div>
+              {copyButton}
+            </div>
+          ) : (
+            <div>
+              <div className="flex justify-between pb-3">
+                {formatting?.introSentence ??
+                getCopyTitle(preference?.formatLanguage) ? (
+                  <div className="text-sm" style={{ paddingTop: "2px" }}>
+                    <span className="color-gray-700">
```
Comment (juliusmarminge) [inline packages/calendar/react/src/components/action-pane/availability.tsx:225]:
```suggestion
                  <div className="text-sm pt-[2px]">
                     <span className="text-gray-700">
```

File: packages/calendar/react/src/components/action-pane/availability.tsx [hunk 21]
```diff
@@ -173,78 +206,90 @@ const Availability: React.FC<{
...
-                        })} ${getTimeZoneAbv(
-                          selectedTimeZone,
-                          convertToDate(start),
-                        )}`}
-                      </span>
-                      <BiTrash
-                        className="ml-3 h-4 w-4 cursor-pointer"
-                        onClick={() => deleteSlot(event)}
-                      />
+                    <div className="text-xs-05 color-gray-700 pb-1" key={idx}>
```
Comment (juliusmarminge) [inline packages/calendar/react/src/components/action-pane/availability.tsx:252]:
```suggestion
                    <div className="text-xs-05 text-gray-700 pb-1" key={idx}>
```

File: apps/extension/src/pages/formatting.tsx [hunk 22]
```diff
@@ -0,0 +1,276 @@
...
+  );
+
+  const introSentenceElem = (
+    <div>
+      <div className="flex border border-gray-100 rounded w-full h-9 overflow-hidden">
+        {Object.values(IntroSentenceType)
+          .map((type) => (
+          <div
+            key={type}
+            className={`flex-1 cursor-pointer text-sm flex items-center justify-center ${
```
Comment (juliusmarminge) [inline apps/extension/src/pages/formatting.tsx:131]: using clsx here will make it more readable

**Review by juliusmarminge — COMMENTED** [review]
Okay, I think this is why your styles aren't applying.

`color-gray-xxx` isn't a valid tailwind class, it should be `text-gray-xxx`
